packages/medusa/src/services/customer-group.ts

```diff
@@ -214,29 +213,34 @@ class CustomerGroupService extends TransactionBaseService {
    * @return the result of the find operation
    */
   async listAndCount(
-    selector: FilterableCustomerGroupProps = {},
+    selector: Partial<CustomerGroup> & {
+      q?: string
+      discount_condition_id?: string
+    } = {},
     config: FindConfig<CustomerGroup>
   ): Promise<[CustomerGroup[], number]> {
     const cgRepo: CustomerGroupRepository = this.manager_.getCustomRepository(
       this.customerGroupRepository_
     )
 
     let q
-    if ("q" in selector) {
+    if ("q" in selector && selector.q) {
```

Comment:
nit: `in` seems to be a weaker check than `selector.q` since `in` will return true even if `selector.q` is defined as `undefined`. 

If there is not specific reason to keep it I think we should just check `selector.q` or `typeof selector.q !== 'undefined'`